firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -925,13 +934,24 @@ private void applyFieldAnnotations(Field field) {
         serverTimestamps.add(propertyName(field));
       }
 
-      if (field.isAnnotationPresent(DocumentId.class)) {
+      if (field.isAnnotationPresent(DocumentId.class)
+          || isAnnotationPresentOnKtxGetters(field, DocumentId.class)) {
         Class<?> fieldType = field.getType();
         ensureValidDocumentIdType("Field", "is", fieldType);
         documentIdPropertyNames.add(propertyName(field));
       }
     }
 
+    // Ktx only apply annotations on top of getters: public void getXxx$annotations()
+    private boolean isAnnotationPresentOnKtxGetters(
+        Field field, Class<? extends Annotation> annotationType) {
+      String fieldName = propertyName(field);
+      if (ktxGeneratedGetters.containsKey(fieldName)) {
+        return ktxGeneratedGetters.get(fieldName).isAnnotationPresent(annotationType);
+      }
+      return false;
+    }
+
     private void applyGetterAnnotations(Method method) {
       if (method.isAnnotationPresent(ServerTimestamp.class)) {
```

Comment:
hm, I don't understand this code too well, but it seems unnecessary to have a dedicated kotlin getter map(above). how about just modifying `applyGetterAnnotations()` with something like the following:



```suggestion
    private static boolean isGetterAnnotationPresent(
        Method method,
        Class<? extends Annotation> annotation) {
      if (method.isAnnotationPresent(annotation)) { return true; }
      try {
        Method getterAnnotationsMethod = method
            .getClass()
            .getMethod(method.getName() + "$annotations");
            return method.isSynthetic() && method.isAnnotationPresent(annotation);
      } catch (NoSuchMethodException ex) {
        return false;
      }
    }
    private void applyGetterAnnotations(Method method) {
      if (isGetterAnnotationPresent(method, ServerTimestamp.class)) {
```

And the same for DocumentId below. this should be the only change required afaict.

But we should definitely check with someone on the firestore team as well.

Reply:
Thank you for this code, it works with some minor changes. This is way shorter than my `applyFieldAnnotations` version, thank you.